### ISSUE TITLE
Implement LayerNorm in tfjs-layers

### ISSUE THREAD

**Issue by aihuawu** [tensorflow/tfjs#1548]
def layer_norm(input_tensor, name=None):
  """Run layer normalization on the last dimension of the tensor."""
  return tf.contrib.layers.layer_norm(
      inputs=input_tensor, begin_norm_axis=-1, begin_params_axis=-1, scope=name)

how can i use layer_norm in js?
Thanks.

george wu

**Comment by caisq** [comment]
There is a `LayerNormalization` layer type in tensorflow.kears (Python). The source code is at 
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/layers/normalization.py#L807

Perhaps the best way to go is to port this layer to TF.js? This layer isn't so complicated (less complicated than BatchNormalization). 

@aihuawu I'm curious. What model are you writing in TF.js that requires layer normalization?

**Comment by aihuawu** [comment]
@caisq 

Thank you for the information.
I am doing some NLP training and I am using the BERT model with multi-head self attention.
In that paper, it has a layer_norm for every sublayer. I am not sure which normalization I should use.
But they use layer normalization instead of batch normalization.

george wu

**Comment by caisq** [comment]
cc @annxingyuan about language (BERT) models in TF.js

**Comment by Derek-Wds** [comment]
@caisq Hi, I'm interested in this issue and I would like to help with it, but I guess we may need to discuss this a bit. 

I see that there is another one [#626](https://github.com/tensorflow/tfjs/issues/626), which is about the another form of normalization. It seems that these two kinds of normalization as well as another one: group normalization mentioned [here](https://arxiv.org/pdf/1803.08494.pdf) are all derivatives of basic normalization: the operation are the same but on different dimensions/axes (this is mentioned in He's paper).

Maybe we just modify the batch normalize class so that it could be used by other normalization type. There is a variable `reductionAxes`, I guess we could just change its value so that we could do normalization on different dimensions/axes?

Derek
